### PR TITLE
add third-party license documentation

### PR DESCRIPTION
**Pull request by bobheadxi**
documents the new workflow in https://github.com/sourcegraph/sourcegraph/pull/11321

### CODE REVIEW

**Comment by nicksnyder** [comment]
Thanks for adding docs! These docs feel a little bit unrelated to code review.

What if we created a separate/new handbook page for documentation about CI and the checks we enforce (e.g., `ci.md`)?

To be clear, I am not asking you to document all that, but if you agree then create that file and add the content you already wrote to it.

**Comment by bobheadxi** [comment]
> What if we created a separate/new handbook page for documentation about CI and the checks we enforce (e.g., ci.md)?

Sounds good! I've created `continuous_integration.md` instead for consistency, since there's already a `continuous_releasability.md`

File: handbook/engineering/continuous_integration.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Continuous Integration
+
+We have a variety of tooling on [Buildkite](https://buildkite.com/sourcegraph/sourcegraph) and [GitHub Actions](https://github.com/sourcegraph/sourcegraph/actions) for continuous integration.
+
+- [GitHub Actions](#github-actions)
+  - [Third-Party Licenses](#third-party-licenses)
+
+## GitHub Actions
+
+### Third-Party Licenses
```
Comment (unknwon) [inline handbook/engineering/continuous_integration.md:10]: nit: we use sentence case
```diff
-### Third-Party Licenses
+### Third-party licenses
```

File: handbook/engineering/continuous_integration.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Continuous Integration
```
Comment (unknwon) [inline handbook/engineering/continuous_integration.md:1]:
```diff
-# Continuous Integration
+# Continuous integration
```

**Pull request merged**